kotlin/android/app/src/main/java/dev/firezone/android/features/splash/ui/SplashViewModel.kt

```diff
@@ -32,15 +33,14 @@ internal class SplashViewModel
                 if (!hasVpnPermissions(context)) {
                     actionMutableLiveData.postValue(ViewAction.NavigateToVpnPermission)
                 } else {
-                    repo.getToken().collect {
-                        if (it.isNullOrBlank()) {
-                            actionMutableLiveData.postValue(ViewAction.NavigateToSignIn)
-                        } else {
-                            // token will be re-read by the TunnelService
-                            if (!TunnelService.isRunning(context)) TunnelService.start(context)
-
-                            actionMutableLiveData.postValue(ViewAction.NavigateToSession)
-                        }
+                    val token = applicationRestrictions.getString("token") ?: repo.getTokenSync()
+                    if (token.isNullOrBlank()) {
+                        actionMutableLiveData.postValue(ViewAction.NavigateToSignIn)
+                    } else {
+                        // token will be re-read by the TunnelService
+                        if (!TunnelService.isRunning(context)) TunnelService.start(context)
+
+                        actionMutableLiveData.postValue(ViewAction.NavigateToSession)
```

Comment:
Thanks for the refactor! There were some strange behaviors on the splash screen with an invalid token when it was being stored in the applicationRestrictions.

With this change, it should be easier to manage different behavior between the two.